Source/WebGPU/WebGPU/Queue.swift

```diff
@@ -30,33 +30,37 @@ private let largeBufferSize = 32 * 1024 * 1024
 public func writeBuffer(
     queue: WebGPU.Queue, buffer: WebGPU.Buffer, bufferOffset: UInt64, data: SpanUInt8
 ) {
-    unsafe queue.writeBuffer(buffer, bufferOffset: bufferOffset, data: data)
+    unsafe queue.writeBuffer(buffer.buffer(), bufferOffset: bufferOffset, data: data)
 }
 
 extension WebGPU.Queue {
-    public func writeBuffer(_ buffer: WebGPU.Buffer, bufferOffset: UInt64, data: SpanUInt8) {
-        let device = unsafe self.device()
-        guard let blitCommandEncoder = ensureBlitCommandEncoder() else {
+    public func writeBuffer(_ buffer: MTLBuffer, bufferOffset: UInt64, data: SpanUInt8) {
+        guard let device = self.metalDevice() else {
             return
         }
-        let noCopy = unsafe data.size() >= largeBufferSize
-        guard device.device() != nil else {
+
+        guard let blitCommandEncoder = ensureBlitCommandEncoder() else {
             return
         }
-        guard
-            let tempBuffer =
-                unsafe noCopy
-                ? device.newBufferWithBytesNoCopy(
-                    data.__dataUnsafe(), data.size(), MTLResourceOptions.storageModeShared)
-                : device.newBufferWithBytes(
-                    data.__dataUnsafe(), data.size(), MTLResourceOptions.storageModeShared)
-        else {
+
+        let noCopy = unsafe data.size() >= largeBufferSize
```

Comment:
I'm not sure how to reduce `unsafe` usages here. Even calling `.size()` I am seeing a warning that the call is unsafe :(

Reply:
I made a minimal test case for `std::span::size` being unsafe and filed rdar://159839254. We'll see what comes back!

Reply:
Any operation that touches a `std::span` will be unsafe since it's an unsafe type (it doesn't carry lifetime information), so even if it's just accessing a property it doesn't matter. I think it's ok to land as is, this file still needs to be reworked, we might have need to tweak the API to reduce the unsafeness.